azure-quantum/azure/quantum/qiskit/backends/rigetti.py

```diff
@@ -0,0 +1,117 @@
+
+if TYPE_CHECKING:
+    from azure.quantum.qiskit import AzureQuantumProvider
+
+import logging
+logger = logging.getLogger(__name__)
+
+__all__ = [
+    "RigettiSimulatorBackend"
+    "RigettiQPUBackend"
+]
+
+class RigettiBackend(AzureBackend):
+    """Base class for interfacing with a Rigetti backend in Azure Quantum"""
+
+    @classmethod
+    def _default_options(cls):
+        return Options(count=500, entryPoint="main", arguments=[], targetCapability="BaseProfileExecution")
+
+    @classmethod
+    def _azure_config(cls):
+        return {
+            "blob_name": "inputData",
+            "content_type": "qir.v1",
+            "provider_id": "rigetti",
+            "input_data_format": "qir.v1",
+            "output_data_format": "microsoft.quantum-results.v1",
+        }
+
+
+class RigettiSimulatorBackend(RigettiBackend):
+    backend_names = (
+        RigettiTarget.QVM.value,
+    )
+
+    def __init__(
+        self,
+        name: str,
+        provider: "AzureQuantumProvider",
+        **kwargs
+    ):
+        """Base class for interfacing with an Rigetti Simulator backend"""
+        default_config = BackendConfiguration.from_dict(
+            {
+                "backend_name": name,
+                "backend_version": __version__,
+                "simulator": True,
+                "local": False,
+                "coupling_map": None,
+                "description": "Rigetti simulator on Azure Quantum",
+                "basis_gates": QIR_BASIS_GATES,
+                "memory": False,
+                "n_qubits": 38,
+                "conditional": False,
+                "max_shots": 10000,
+                "max_experiments": 1,
+                "open_pulse": False,
+                "gates": [{"name": "TODO", "parameters": [], "qasm_def": "TODO"}],
+                "azure": self._azure_config(),
+            }
+        )
+        logger.info("Initializing RigettiSimulatorBackend")
+        configuration: BackendConfiguration = kwargs.pop("configuration", default_config)
+        super().__init__(configuration=configuration, provider=provider, **kwargs)
+
+
+class RigettiQPUBackend(RigettiBackend):
+    backend_names = (
+        RigettiTarget.ASPEN_11.value,
+        RigettiTarget.ASPEN_M_1.value
+    )
+
+    def __init__(
+        self,
+        name: str,
+        provider: "AzureQuantumProvider",
+        **kwargs
+    ):
+        """Base class for interfacing with a Rigetti QPU backend"""
+        default_config = BackendConfiguration.from_dict(
+            {
+                "backend_name": name,
+                "backend_version": __version__,
+                "simulator": False,
+                "local": False,
+                "coupling_map": None,
+                "description": "Rigetti QPU on Azure Quantum",
+                "basis_gates": QIR_BASIS_GATES,
+                "memory": False,
+                "n_qubits": 80 if name == RigettiTarget.ASPEN_M_1.value else 38,
```

Comment:
Perhaps this should also be a function or similar on `RigettiTarget` so it's more clear that it needs to be updated when new QPUs are added.